### PR TITLE
Remove grpc from opa-connector

### PR DESCRIPTION
**Pull request by shlomitk1**
Signed-off-by: Shlomit Koyfman <shlomitk@il.ibm.com>
Co-authored by Roee Shlomo

Fixes https://github.com/fybrik/fybrik/issues/1062

### CODE REVIEW

**Review by roee88 — COMMENTED** [review]
The katalog connector accepts arguments for he bind address which is useful when running locally (e.g., to specify another port). You may want to do the same here. 

We probably need another reviewer on this PR since we both worked on it.

File: connectors/opa/connector.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		return
+	}
+	// Get asset metadata from catalog connector
+	requestToCatalog := &datacatalog.GetAssetRequest{
+		AssetID:       request.Resource.ID,
+		OperationType: datacatalog.READ,
+	}
+	assetInfo, err := r.CatalogClient.GetAssetInfo(requestToCatalog, c.GetHeader(headerCredentials))
+	if err != nil {
+		// TODO: better error propagation
```
Comment (Mohammad-nassar10) [inline connectors/opa/connector.go:54]: Is this TODO for this PR or in the future?

Comment (shlomitk1) [reply connectors/opa/connector.go:54]: future

**Comment by shlomitk1** [comment]
> The katalog connector accepts arguments for the bind address which is useful when running locally (e.g., to specify another port). You may want to do the same here.
> 
> We probably need another reviewer on this PR since we both worked on it.

Correct but these arguments are also hard-coded: https://github.com/fybrik/fybrik/blob/7773f07f7bebc5b59af79f7b2837165264257ce6/connectors/katalog/main.go#L32-L33

**Comment by roee88** [comment]
> > The katalog connector accepts arguments for the bind address which is useful when running locally (e.g., to specify another port). You may want to do the same here.
> 
> Correct but these arguments are also hard-coded:
> 
> https://github.com/fybrik/fybrik/blob/7773f07f7bebc5b59af79f7b2837165264257ce6/connectors/katalog/main.go#L32-L33

They are not, these are the defaults. They are overridden by command line arguments.

File: connectors/opa/main.go [hunk 2]
```diff
@@ -1,71 +1,101 @@
...
+func NewRouter(controller *ConnectorController) *gin.Engine {
+	router := gin.Default()
+	router.POST("/getPoliciesDecisions", controller.GetPoliciesDecisions)
+	return router
+}
+
+// RootCmd defines the root cli command
+func RootCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "opa",
```
Comment (roee88) [inline connectors/opa/main.go:36]: opa-connector probably

File: connectors/opa/main.go [hunk 3]
```diff
@@ -1,71 +1,101 @@
...
-func main() {
-	log.Println("in main of V2 OPA Connector: ")
-	catalogConnectorAddress := getEnv("CATALOG_CONNECTOR_URL")
-	catalogProviderName := getEnv("CATALOG_PROVIDER_NAME")
+// RunCmd defines the command for running the connector
+func RunCmd() *cobra.Command {
+	ip := ""
+	port := 8080
+	cmd := &cobra.Command{
+		Use:   "run",
```
Comment (roee88) [inline connectors/opa/main.go:48]: Need to add to dockerfile `CMD [ "run" ]`

**Review by roee88 — APPROVED** [review]

File: connectors/opa/main.go (hunk 2 above)
Comment (shlomitk1) [reply connectors/opa/main.go]: changed

File: connectors/opa/main.go (hunk 3 above)
Comment (shlomitk1) [reply connectors/opa/main.go:48]: added

**Review by Mohammad-nassar10 — APPROVED** [review]

**Pull request merged**